plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.tsx

```diff
@@ -96,14 +95,12 @@ export function EntityAutocompletePicker<
 
   // Set selected options on query parameter updates; this happens at initial page load and from
   // external updates to the page location
-  useEffect(() => {
-    if (
-      queryParameters.length &&
-      !_.isEqual(selectedOptions, queryParameters)
-    ) {
-      setSelectedOptions(queryParameters);
-    }
-  }, [selectedOptions, queryParameters]);
+  const [prevQueryParameters, setQueryParameter] = useState(queryParameters);
+  // if the query parameter has changed, update the selected options
+  if (queryParameters !== prevQueryParameters) {
+    setSelectedOptions(queryParameters);
+    setQueryParameter(queryParameters);
+  }
```

Comment:
Mutating state directly on render should be avoided, we'll want to keep this wrapped in `useEffect` or some other solution. Does simply removing the `selectedOptions` from the existing `useEffect` work?

Reply:
@headphonejames are you able to update this PR to align with the requested changes?

Reply:
I updated the change and verified that it is working.